src/DoubleTheDonation/API/REST/CompanyMatching.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace GiveDoubleTheDonation\DoubleTheDonation\API\REST;
+
+use Give\Donations\Models\Donation;
+use GiveDoubleTheDonation\DoubleTheDonation\FormExtension\Actions\FieldScope\HandleData;
+use WP_REST_Request;
+use WP_REST_Server;
+
+/**
+ * @unreleased
+ */
+class CompanyMatching
+{
+    public function __invoke()
+    {
+        register_rest_route(
+            'givewp/dtd',
+            'donation/(?P<id>[0-9]+)',
+            [
+                [
+                    'methods' => WP_REST_Server::EDITABLE,
+                    'callback' => function (WP_REST_Request $request) {
+                        $data = [
+                            'company_id' => $request->get_param('companyId'),
+                            'company_name' => $request->get_param('companyName'),
+                            'entered_text' => $request->get_param('enteredText')
+                        ];
+
+                        $donation = Donation::find($request->get_param('id'));
+
+                        $donationHandler = new HandleData();
+                        $donationHandler->save($data, $donation);
+                        $donationHandler->send($data, $donation);
+                    },
+                    'permission_callback' => function(WP_REST_Request $request) {
+                        return (bool)give()->donations->getByReceiptId($request->get_param('receiptId'));
```

Comment:
@alaca actually one more thing I think we should do here is make sure the receipt ID belongs to the donation for this request so something like:

```suggestion
$receiptId = $request->get_param('receiptId');
$donationId = $request->get_param('id');
$donation = give()->donations->getByReceiptId($receiptId);

if (!$donation) {
    return false;
}

return $donation->id === $donationId;
```

Reply:
Great suggestion. Thanks, man. Resolved: https://github.com/impress-org/give-double-the-donation/pull/25/commits/e3703d6f52be31c1c8baac57055165843491e2fb